numpy/testing/_private/pytesttester.py

```diff
@@ -0,0 +1,329 @@
+
+      from numpy.testing import PytestTester
+      test = PytestTester().test
+
+    Calling this test function finds and runs all tests associated with the
+    package and all its sub-packages.
+
+    Attributes
+    ----------
+    package_path : str
+        Full path to the package to test.
+    package_name : str
+        Name of the package to test.
+
+    Parameters
+    ----------
+    package : module, str or None, optional
+        The package to test. If a string, this should be the full path to
+        the package. If None (default), `package` is set to the module from
+        which `PytestTester` is initialized.
+    raise_warnings : None, str or sequence of warnings, optional
+        This specifies which warnings to configure as 'raise' instead
+        of being shown once during the test execution.  Valid strings are:
+
+          - "develop" : equals ``(Warning,)``
+          - "release" : equals ``()``, don't raise on any warnings.
+
+        Default is "release".
+    depth : int, optional
+        If `package` is None, then this can be used to initialize from the
+        module of the caller of (the caller of (...)) the code that
+        initializes `NoseTester`. Default of 0 means the module of the
+        immediate caller; higher values are useful for utility routines that
+        want to initialize `NoseTester` objects on behalf of other code.
+
+    """
+    def __init__(self, package=None, raise_warnings="release", depth=0,
+                 check_fpu_mode=False):
+        # Back-compat: 'None' used to mean either "release" or "develop"
+        # depending on whether this was a release or develop version of numpy.
+        # Those semantics were fine for testing numpy, but not so helpful for
+        # downstream projects like scipy that use numpy.testing. (They want to
+        # set this based on whether *they* are a release or develop version,
+        # not whether numpy is.) So we continue to accept 'None' for
+        # back-compat, but it's now just an alias for the default "release".
+        if raise_warnings is None:
+            raise_warnings = "release"
+
+        package_name = None
+        if package is None:
+            frame = sys._getframe(1 + depth)
+            package_path = frame.f_locals.get('__file__', None)
+            if package_path is None:
+                raise AssertionError
+            package_path = os.path.dirname(package_path)
+            package_name = frame.f_locals.get('__name__', None)
+        elif isinstance(package, type(os)):
+            package_path = os.path.dirname(package.__file__)
+            package_name = getattr(package, '__name__', None)
+        else:
+            package_path = str(package)
+
+        self.package_path = package_path
+
+        # Find the package name under test; this name is used to limit coverage
+        # reporting (if enabled).
+        if package_name is None:
+            package_name = _get_package_name(package_path)
```

Comment:
I'd be tempted to combine this with the `else` above

Reply:
We can cure that problem if we decide to simply remove the function.